clang/lib/CConv/Utils.cpp

```diff
@@ -384,3 +384,27 @@ unsigned int getParameterIndex(ParmVarDecl *PV, FunctionDecl *FD) {
   }
   llvm_unreachable("Parameter declaration not found in function declaration.");
 }
+
+bool evaluateToInt(Expr *E, const ASTContext &C, int &Result) {
+  Expr::EvalResult ER;
+  E->EvaluateAsInt(ER, C, clang::Expr::SE_NoSideEffects, false);
+  if (ER.Val.isInt()) {
+    Result = ER.Val.getInt().getExtValue();
+    return  true;
+  }
+  return false;
+}
+
+bool isZeroBoundsExpr(BoundsExpr *BE, const ASTContext &C) {
+  if (auto *CBE = dyn_cast<CountBoundsExpr>(BE)){
+    // count(0) and byte_count(0)
+    Expr *E = CBE->getCountExpr();
+    int Result;
+    if (evaluateToInt(E, C, Result))
+      return Result == 0;
```

Comment:
We have the function `Utils::isNULLExpression` -- how does it relate to the functions you've added? Is this an opportunity to combine them ?

Reply:
`isNULLExpression` is much more specific in terms of what it accepts. It makes use of a clang function `isNullPointerConstant` for the test where as `isZeroBoundsExpr` evaluates the expression to see if it evaluates to `0`. We could possibly replace `isZeroBoundsExpr`

Reply:
I thought that `isNullPointerConstant` also basically evaluated the result to zero ? If not, then we can leave things as is.

Reply:
I think they're similar enough that we could probably get away with it, but it's not a change I really want to make since there are differences.